### PR TITLE
[Resources] Add `Publish-AzBicepModule` for publishing Bicep files

### PR DESCRIPTION
**Pull request by shenglol**
## Description

In the recent release of Bicep CLI, we added a new command `bicep publish` for publish a module to a container registry (see https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-cli#publish). The PR adds a new cmdlet `Publish-AzBicepModule` which allows Bicep and Azure PowerShell users to publish Bicep modules via PowerShell without the need to invoke the `bicep publish` command.

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by BethanyZhou — DISMISSED** [review]

File: src/Resources/ResourceManager/Implementation/Bicep/PublishAzureBicepModuleCmdlet.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Utilities;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.Bicep
+{
+    [Cmdlet(VerbsData.Publish, AzureRMConstants.AzureRMPrefix + "BicepModule", SupportsShouldProcess = true), OutputType(typeof(void))]
```
[the remark this answers is not in the document]
Comment (shenglol) [reply src/Resources/ResourceManager/Implementation/Bicep/PublishAzureBicepModuleCmdlet.cs:8]: Done.

File: src/Resources/ResourceManager/Utilities/BicepUtility.cs [hunk 2]
```diff
@@ -12,132 +12,145 @@
...
+
             powershell.AddScript("Get-Command bicep");
             powershell.Invoke();
             powershell.AddScript("$?");
             var result = powershell.Invoke();
-            bool.TryParse(result[0].ToString(), out bool res);
             // Cache result
-            IsBicepExecutable = res;
-            return IsBicepExecutable;
+            bool.TryParse(result[0].ToString(), out IsBicepExecutable);
```
[the remark this answers is not in the document]
Comment (shenglol) [reply src/Resources/ResourceManager/Utilities/BicepUtility.cs]: Sure thing. Verified that the value does get cached.

File: src/Resources/Resources/help/Publish-AzBicepModule.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+```
+
+## DESCRIPTION
+The **Publish-AzBicepModule** cmdlet publishes a Bicep file to a registry as a module. To use the cmdlet, Bicep CLI (version 0.4.1008 or later) must be installed and added to PATH.
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:\> Publish-AzBicepModule -File main.bicep -Target br:{registry}/{moduleName}:{tag}
```
Comment (isra-fel) [inline src/Resources/Resources/help/Publish-AzBicepModule.md:27]:
```suggestion
PS C:\> Publish-AzBicepModule -FilePath main.bicep -Target br:{registry}/{moduleName}:{tag}
```

File: src/Resources/Resources/help/Publish-AzBicepModule.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+Required: True
+Position: Named
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### -PassThru
+{{ Fill PassThru Description }}
```
Comment (isra-fel) [inline src/Resources/Resources/help/Publish-AzBicepModule.md:65]: Indicates that this cmdlet returns a boolean result. By default, this cmdlet does not generate any output.

File: src/Resources/ResourceManager/Implementation/Bicep/PublishAzureBicepModuleCmdlet.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Local path to the bicep file to publish.")]
+        [ValidateNotNullOrEmpty]
+        public string FilePath { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The target location where the bicep file will be published.")]
+        [ValidateNotNullOrEmpty]
+        public string Target { get; set; }
+
+        [Parameter(Mandatory = false)]
```
Comment (isra-fel) [inline src/Resources/ResourceManager/Implementation/Bicep/PublishAzureBicepModuleCmdlet.cs:19]:
```suggestion
        [Parameter(Mandatory = false, HelpMessage="Indicates that this cmdlet returns a boolean result. By default, this cmdlet does not generate any output.")]
```
And then regenerated help

File: src/Resources/ResourceManager/Implementation/Bicep/PublishAzureBicepModuleCmdlet.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Utilities;
```
Comment (isra-fel) [inline src/Resources/ResourceManager/Implementation/Bicep/PublishAzureBicepModuleCmdlet.cs:1]: License header please

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/Resources/ResourceManager/Implementation/Bicep/PublishAzureBicepModuleCmdlet.cs (hunk 6 above)
Comment (shenglol) [reply src/Resources/ResourceManager/Implementation/Bicep/PublishAzureBicepModuleCmdlet.cs]: Good catch. Added license header.

**Comment by shenglol** [comment]
@isra-fel I've addressed the comments. Mind taking another look?

**Review by isra-fel — APPROVED** [review]
LGTM

File: src/Resources/Resources/help/Publish-AzBicepModule.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
```
Comment (BethanyZhou) [inline src/Resources/Resources/help/Publish-AzBicepModule.md:65]: Please replace this placeholder by regenerating the help file.

**Review by isra-fel — DISMISSED** [review]

**Review by isra-fel — DISMISSED** [review]
LGTM

File: src/Resources/ResourceManager/Implementation/Bicep/PublishAzureBicepModuleCmdlet.cs [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Utilities;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.Bicep
+{
+    [Cmdlet(VerbsData.Publish, AzureRMConstants.AzureRMPrefix + "BicepModule", SupportsShouldProcess = true), OutputType(typeof(void))]
```
Comment (BethanyZhou) [inline src/Resources/ResourceManager/Implementation/Bicep/PublishAzureBicepModuleCmdlet.cs:22]: LGTM. Output type should be Boolean and its output in docs should be updated as well.

Comment (shenglol) [reply src/Resources/ResourceManager/Implementation/Bicep/PublishAzureBicepModuleCmdlet.cs]: Updated output type in source code and docs.

**Review by isra-fel — APPROVED** [review]

**Pull request merged**